pkg/volume/metrics_du.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volume
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+var _ VolumeMetricsProvider = &MetricsDu{}
+
+// MetricsDu represents a VolumeMetricsProvider that calculates the used and available
+// Volume capacity by executing the "du" and "df" commands on the Volume path.
+type MetricsDu struct {
+	// the directory path the volume is mounted to.
+	path string
+}
+
+// InitMetricsDu initializes MetricsDu with the Volume path.
+// This is required before MetricsDu can be used.
+func (cm *MetricsDu) InitMetricsDu(path string) {
+	cm.path = path
+}
+
+// See VolumeMetricsProvider.GetCapacityMetrics
+// GetCapacityMetrics calculates the volume usage and device free space by executing "du"
+// and "df" on path.
+func (cm *MetricsDu) GetCapacityMetrics() (*CapacityMetrics, error) {
```

Comment:
The interface doesn't have `Capacity` in it now, but I can drop it from the method / struct.  Wasn't sure if we would want to break metrics of other types out separately or bundle them together.  I'll update it per your suggestion and see how it looks.